src/Engine/EngineInterface.php

```diff
@@ -10,6 +10,15 @@
 
 interface EngineInterface
 {
+
+    /**
+     * Requires the Analytics instance which handles the Events
+     * that occur during the test process.
+     * 
+     * @param AnalyticsInterface $analytics
+     */
+    public function __construct(AnalyticsInterface $analytics);
```

Comment:
That's not what I meant at least :) The contract already enforces the implementation to have an instance of AnalyticsInterface since there is an `getAnalytics` method...